0-module-and-module-source.emu

```diff
@@ -0,0 +1,650 @@
+              ): ~unused~
+            </h1>
+            <dl class="header">
+              <dt>description</dt>
+              <dd>FinishModuleDynamicImport completes the process of a dynamic import originally started by an <emu-xref href="#sec-import-calls">`import()`</emu-xref> call for a Module Instance, resolving or rejecting the promise returned by that call as appropriate according to _innerPromise_'s resolution. It is performed by host environments as part of HostImportModuleRecordDynamically.</dd>
+            </dl>
+            <emu-alg>
+              1. Let _fulfilledClosure_ be a new Abstract Closure with parameters (_result_) that captures _moduleRecord_ and _promiseCapability_ and performs the following steps when called:
+                1. Assert: _result_ is *undefined*.
+                1. Assert: Evaluate has already been invoked on _moduleRecord_ and successfully completed.
+                1. Let _namespace_ be Completion(GetModuleNamespace(_moduleRecord_)).
+                1. If _namespace_ is an abrupt completion, then
+                  1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+                1. Else,
+                  1. Perform ! Call(_promiseCapability_.[[Resolve]], *undefined*, &laquo; _namespace_.[[Value]] &raquo;).
+                1. Return ~unused~.
+              1. Let _onFulfilled_ be CreateBuiltinFunction(_fulfilledClosure_, 0, *""*, &laquo; &raquo;).
+              1. Let _rejectedClosure_ be a new Abstract Closure with parameters (_error_) that captures _promiseCapability_ and performs the following steps when called:
+                1. Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; _error_ &raquo;).
+                1. Return ~unused~.
+              1. Let _onRejected_ be CreateBuiltinFunction(_rejectedClosure_, 0, *""*, &laquo; &raquo;).
+              1. Perform PerformPromiseThen(_innerPromise_, _onFulfilled_, _onRejected_).
+              1. Return ~unused~.
+            </emu-alg>
+
+            <emu-note type="editor">
+                This Abstract Operation is analogous to FinishDynamicImport but for Module Records with an associated Module Instance.
+            </emu-note>
+          </emu-clause>
+    </emu-clause>
+
+    <emu-clause id="sec-module-constructor">
+        <h1>The Module Constructor</h1>
+        <p>The Module constructor:</p>
+        <ul>
+            <li>is the intrinsic object <dfn>%Module%</dfn>.</li>
+            <li>is the initial value of the *"Module"* property of the global object.</li>
+            <li>is not intended to be called as a function and will throw an exception when called in that manner.</li>
+            <li>creates and initializes a new Module object when called as a constructor.</li>
+        </ul>
+
+        <emu-clause id="sec-module">
+            <h1>Module ( _moduleSource_, _referral_, _importHook_ [, _importMeta_ ] )</h1>
```

Comment:
`importHook` should be optional argument. Inherit the current evaluator's import hook.

```js
const x = new Module(source, 'spec') // uses host
const ev = new Evaluators({ importHook: myHook })
new ev.Module(source, 'spec') // uses myHook
```

Reply:
Resolution: we will look into the default hook in another PR/issue.